swupd/packs_test.go

```diff
@@ -377,6 +378,29 @@ func mustValidateZeroPack(t *testing.T, manifestPath, packPath string) {
 		_ = tr.Close()
 	}()
 
+	mustHaveDir := func(name string) {
+		hdr, err := tr.Next()
+		if err == io.EOF {
+			t.Fatalf("invalid pack: required dir %s not found", name)
+		}
+		if err != nil {
+			t.Fatalf("error reading pack: %s", err)
+		}
+		if hdr.Name != name {
+			t.Fatalf("invalid pack: required dir %s not found", name)
+		}
+		if hdr.Typeflag != tar.TypeDir {
+			t.Fatalf("invalid pack: %s is of type %c instead of %c (directory)", name, hdr.Typeflag, tar.TypeDir)
+		}
+		var expectedMode int64 = 0700
+		if hdr.Mode != expectedMode {
+			t.Fatalf("invalid pack: wrong permissions %s for %s, expected %s", os.FileMode(hdr.Mode), name, os.FileMode(expectedMode))
+		}
+	}
+
+	mustHaveDir("delta/")
```

Comment:
These rely on order right? If I'm right, can you put a comment in?

Reply:
Done. Also mentioned I think we can relax this in the future. For now I think the validation function is fine as is.